nixos/modules/virtualisation/keystone.nix

```diff
@@ -0,0 +1,211 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.virtualisation.keystone;
+  keystoneConf = pkgs.writeText "keystone.conf" ''
+    [DEFAULT]
+    admin_token = ${cfg.adminToken}
+    policy_file=${cfg.package}/etc/policy.json
+
+    [database]
+    connection = mysql://${cfg.database.user}:${cfg.database.password}@${cfg.database.host}/${cfg.database.name}
```

Comment:
I'm not sure, but maybe would be better to move all this config line as parameter instead of splitting it into smaller parts? For example keystone supports postgresql as well, in such a case user could just paste all the connection string.

In the description field there could be a link to keystone documentation. there is also example field in mkOption so you could provide one or two examples.

And probably this could be a "secret".